BC5 updates November

MoteHue opened the pull request.

MoteHue committed:
Reduce memory cap on cluster nodes

MoteHue committed:
Give demo_user sudo rights

MoteHue committed:
Extra filesystem mounts

MoteHue committed:
Set timezone for cluster

MoteHue committed:
Staging updates

MoteHue committed:
Run timezone hook in builder

MoteHue committed:
Mount /tmp under tmpfs

Also includes an adhoc playbook, used to apply this on the running
cluster without the need for an image rebuild.

milliams commented on environments/production/inventory/group_vars/all/basic_users.yml:
```diff
@@ -3,3 +3,4 @@ basic_users_users:
     password: "{{ vault_demo_user_password | password_hash('sha512', 65534 | random(seed=inventory_hostname) | string) }}" # idempotent
     uid: 1005
     create_home: false
+    sudo: demo_user ALL=(ALL) NOPASSWD:ALL
```
We probably don't want `demo_user` to have passwordless sudo. In fact, we probably want to remove the user entirely.

MoteHue replied:
Agreed, this is a remnant from before proper user access was configured.

milliams commented on environments/site/inventory/group_vars/all/basic_users.yml:
```diff
@@ -3,6 +3,7 @@ basic_users_users:
     password: "{{ vault_demo_user_password | password_hash('sha512', 65534 | random(seed=inventory_hostname) | string) }}" # idempotent
     create_home: false
     uid: 1005
+    sudo: demo_user ALL=(ALL) NOPASSWD:ALL
```
As above.

milliams commented on environments/site/inventory/groups:
```diff
@@ -120,7 +120,7 @@ compute
 # Hosts to install specified additional packages on
 builder
-cluster
+#cluster
```
Just for my understanding, this says to only run the `extra_packages` stuff on Packer build, but not when running Ansible directly? Is this an intentionally permanent change?

MoteHue replied:
Yes, this should be a permanent change. 
We were previously installing packages at runtime to allow for faster testing, but now we're getting closer to "proper" production we should instead look to use new images for new package installs.

milliams commented on environments/staging/tofu/main.tf:
```diff
@@ -5,7 +5,8 @@ variable "environment_root" {
     source = "../../site/tofu/"
-
+    
+    cluster_image_id = "48325752-dec3-475c-a6c5-cf41b6c009d0"   # openhpc-251111-0940-93ad86a2
```
Is this a temporary change for testing a new image?

MoteHue replied:
Yes, this is an image I built for testing the tmpfs change via the image, rather than the adhoc playbook. 
We'll likely build a newer image again this afternoon

MoteHue committed:
Store Tofu state in S3

jcwomack committed:
Set DefaultTime to 15min for all partitions (in staging, production)

MoteHue committed:
Remove cluster as child of extra_packages group

... so package install occurs only during image build

jcwomack committed:
Explicitly set availability zone for staging cluster

jcwomack committed:
Configure demo_user for staging cluster

MoteHue committed:
Allow demo_user to submit jobs on staging

Moves the site openhpc.yml, to avoid Ansible variable precedence
getting in the way.

Without this move, environments/site/inventory/group_vars/openhpc.yml
would take precedence over
environments/site/inventory/group_vars/openhpc.yml

milliams committed:
Remove sudo access from demo_user in production

Keep it in staging for now.

The demo_user account will be removed entirely from production at
some point anyway.

milliams commented:
I'm happy with the PR and am happy to have it merged to avoid living on this branch forever :) Any reason to hold it up? I see the CI is failing, but I guess that's due to the upstream repo not being synced in yet?

MoteHue commented:
> I'm happy with the PR and am happy to have it merged to avoid living on this branch forever :) Any reason to hold it up? I see the CI is failing, but I guess that's due to the upstream repo not being synced in yet?

I see no reason to keep it open, I'll merge. 
The CI jobs here aren't needed on your site, we should remove them. These are designed to test changes in the dev work upstream
